<!DOCTYPE html>
<html>
<head>
    <link href="/mysite/static/css/bootstrap.css" rel="stylesheet" type="text/css">
    <link rel="stylesheet" type="text/css" href="{% static "css/bootstrap.css" %}"> 
    <style>
        div.header{
            padding: 0;
            color: white;
            margin-bottom:2px;
            width:100%;
            background-color: #3F3F3F;
            clear: left;
            text-align: center;
            overflow: auto;
        }

        body {
            margin: 0;
            
        }

        ul2 {
            list-style-type: none;
            margin-top:0;
            padding: 1px;
            width: 12%;
            height: 100%;
            background-color: darkgray;
            overflow: auto;
        }

        li a {
            display: block;
            color: white;
            padding: 10px 8px 8px 16px;
            text-decoration: none;
        }

        li a.active {
            background-color: mediumaquamarine;
            color: white;
        }

        li a:hover:not(.active) {
            background-color: mediumpurple;
            color: white;
        }

        ul {
            list-style-type: none;
            margin: 0;
            padding: 0;
            overflow: hidden;
            background-color: #333;
        }

        li2 {
            float: left;
        }

        li2 a {
            display: block;
            color: indigo;
            text-align: center;
            padding: 14px 16px;
            text-decoration: none;
        }


            .container a {
                float: left;
                font-size: 16px;
                color: white;
                text-align: center;
                padding: 14px 16px;
                text-decoration: none;
            }        

    </style>



    <div class="header"> 
        <ul>
        <li><a href="http://otulab.unl.edu:9000/">Home</a></li>
        <li><a class="active" href="/static/readme/readme.html">Tutorial</a></li>
        <li><a href="/static/readme/hg38.html"> <b>Human (Hg38)</b> Statistics</a></li>
        <li><a href="/static/readme/mm10.html"> <b>Mouse (Mm10)</b> Statistics</a></li>
        <li><a href="/static/readme/dm3.html"> <b>Fruit Fly (Dm3)</b> Statistics</a></li>
        <li><a href="/static/readme/ce10.html"> <b>Worm (Ce10)</b> Statistics</a></li>
        <li><a href="/static/readme/rn6.html"> <b> Rat (Rn6)</b> Statistics</a></li>
        <li><a href="https://github.com/dicleyalcin/dbCGI"> <b> Git Repository</b> </a></li>
        <!--<li><a href="#cite">Please cite</a></li>-->
        </ul>
    </div> 



    <style>
               table {
                        border-collapse: collapse;
                        width: 80%;
                    }

               th, td {
                        text-align: left;
                        padding: 10px;
                    }

                tr:nth-child(even){background-color: #f2f2f2}

                th {
                        background-color: #FFA500;
                        color: white;
                    }
                                
                 div.container {
                    width: 1000px;
                    margin-left: 200px;
                }
                
                p.one {
                    margin-left: inherit;
                }
                code { 
                    font-family: monospace;
                }
                </style>

<body style='font-family:"Arial"'>
    
        <center><h1>dbCGI Documentation</h1></center>


        <div class="container">
        dbCGI provides embedded gene ID mapper REST API <a href="http://docs.mygene.info/projects/mygene-py/en/latest/"> mygene</a> 
        and MYSQL to query/retrieve CpG island annotation data.</p> 
        Currently available organisms and assemblies are as follows:
        </div>
         <br></br>
                   
                <div class="container">
                   <table>
                        <tr>
                            <th>Organism</th>
                            <th>Assembly</th> 
                        </tr>
                        <tr>
                            <td><i>Homo sapiens</i></td>
                            <td>Dec. 2013 (hg38, GRCh38)</td>
                        </tr>
                        <tr>
                            <td><i>Mus musculus</i></td>
                            <td>Dec. 2011 (mm10, GRCm38)</td>
                        </tr>
                        <tr>
                            <td><i>Drosophila melanogaster</i></td>
                            <td>Apr. 2006 (BDGP R5/dm3)</td>
                        </tr>

                        <tr>
                            <td><i>Caenorhabditis elegans</i></td>
                            <td>Oct. 2010 (Wormbase v.WS220/ce10)</td>
                        </tr>

                        <tr>
                            <td><i>Rattus norvegicus</i></td>
                            <td>Jul. 2014 (RGSC Rnor_6.0/rn6)</td>
                        </tr>                                                
                        </table>    
                </div>
    <br></br>
    <div class="container">
    <!--In the annotation database, there exists CpG islands found from different algorithms. All are provided at the end of the query, depending on which algorithm
    the user wants to work with. Table below shows the criterias for these three CGI detection algorithms (GC content, length and observed/expected ratio) -->
    dbCGI uses the following three algorithms to detect CGIs in the genome. dbCGI output files contain results separately 
    listed based on each of the three algorithms.The table below shows the criteria  used to define CGIs by the three detection 
    algorithms. 

    
    </div>
    <br></br>
   

                <div class="container">
                   <table>
                        <tr>
                            <th>Algorithm</th>
                            <th>G+C%</th>
                            <th>Length</th>
                            <th>Obs/Exp ratio</th> 
                        </tr>
                        <tr>
                            <td>Gardiner-Garden and Frommer</td>
                            <td> >50%</td>
                            <td> >= 200bp </td>
                            <td> >= 0.60 </td> 
                        </tr>
                        <tr>
                            <td>Takai-Jones</td>
                            <td> >55%</td>
                            <td> >= 500bp</td>
                            <td> >= 0.65</td>
                        </tr>
                        <tr>
                            <td>Ponger-Mouchiroud</td>
                            <td> >50%</td>
                            <td> >= 500bp</td>
                            <td> >= 0.60</td>
                        </tr>
                        </table>    
                </div>
        					
       <div class="container">
           <h2>Annotation Strategy</h2>                         
		 			<img src="/mysite/static/images/annotations_scheme.jpg" width="50%" height="50%">
        </div>             
        <div class="container">
            <h2> Input File </h2> 
            dbCGI accepts plain text files as input. There are two different input file types: 
                <ol type="1">
                    <li2>Gene ID File: In this file each line should contain one gene identifier with no separators 
                        like ,/;/:/ etc. The embedded ID mapper accommodates a wide range of gene identifier types. 
                        <a href='idfile.txt'> Here  you can find a sample Gene ID File.</a>
                    </li2>
                    <li2>Chromosomal Location File: In this file each line represents one continuous interval on the genome. 
                        Chromosome number is indicated by ChrN (e.g. Chr1, Chr2, etc.) followed by the start and end positions, 
                        all separated by '-' (e.g. Chr3-736828-1364826). If you want to query the entire chromosome, 
                        do not enter any position information following the chromosome number. 
                        X and Y chromosomes are denoted by ChrX and ChrY, respectively. 
                        <a href='locfile.txt'> Here  you can find a sample Chromosomal Location File.</a>
                    </li2>
                </ol>             
            <!--Input files can be without extension, (.txt is preferred)
            Each line should only contain one gene, with no separators like ,/;/:/ etc.
            Here are two ID and location file input examples for human taxon.-->
            <br></br>
            <style>

            .IDoutput{
                width: 1200px;
                height: 350px;
                float: center;
                background: url('../images/IDoutput.png');
                background-size: 1200px;
                
             }    
             
            .LocOutput{
                width: 1200px;
                height: 125px;
                float: center;
                background: url('../images/LocOutput.png');
                background-size: 1200px;
                
             }                      
            </style>
            <!--
            <div style="width:700px;">
                <div style="width:400px; float:left;">
                    <h3> ID input </h3>   
                    <div class="IDex"></div>
                    </div>
                <div style="width:300px; float:right;">
                    <h3> Location input </h3>
                    <div class="LocEx"></div>
                    </div>
           </div>
           -->
            
        </div>
        <br><br><br>
        <div class="container">
            
            <style>
                boxmodel {
                    background-color: white;
                    width: 400px;
                    padding: 25px;
                    border: 5px solid orangered;
                    
                }
            </style>
            <boxmodel>
                Please make sure you choose the organism and assembly before you upload your file and submit your query.
                </boxmodel>
            
       </div>
       <br></br>
       
       <div class="container">
           <h2>The output</h2>
            The output file contains an annotated list of CGIs. 
            CGI names represent the detection algorithm, chromosome number, and rank; e.g. "TJ_Chr7_CGI 874"
            represents the 874th overall CGI found in the genome that lies on Chromosome 7, according to the 
            Takai-Jones detection algorithm. 
            <br></br>
            For each CGI, the output file lists start and end positions, G+C%, Length, and observed/expected GC ratio. 
            The output files show slight difference in their formats based on the input file type being Gene ID File 
            (<a href='IDresult.csv'> See File </a>) or Chromosomal Location File 
            (<a href='LOCresult.csv'> See File </a>).
            <br></br>
            <!--
            An example output file for one entry and the fields they represent is shown below:

 
           <h3> ID file output </h3>
           <div class="IDoutput"></div>
           <br></br>
           
           <h3> Location file output </h3>
           <div class="LocOutput"></div> 
           -->
                     
           
           
           
           
       
            


</font>
</body>
</head>
</html>
